hail/python/hail/methods/impex.py

```diff
@@ -2972,6 +2972,325 @@ def import_avro(paths, *, key=None, intervals=None):
+                             _intervals=new_partitioner))
+
+    vd = hl.read_matrix_table(os.path.join(checkpoint_path, 'variant_data'),
+                              _intervals=new_partitioner)
+
+    # read site and vqsr data with same intervals for efficient joins
+    site = hl.read_table(site_path, _intervals=new_partitioner)
+    vqsr = hl.read_table(vqsr_path, _intervals=new_partitioner)
+
+    vd = vd.annotate_rows(filters=hl.coalesce(site[vd.locus].filters, hl.empty_set(hl.tstr)))
+
+    # vqsr ref/alt come in normalized individually, so need to renormalize to the dataset ref allele
+    vd = vd.annotate_rows(as_vqsr = hl.dict(vqsr.index(vd.locus, all_matches=True)
+                                            .map(lambda record: (record.alt + vd.alleles[0][hl.len(record.ref):], record.drop('ref', 'alt')))))
+    vd = vd.annotate_globals(tranche_data=tranche.collect(_localize=False),
+                             truth_sensitivity_snp_threshold=truth_sensitivity_snp_threshold,
+                             truth_sensitivity_indel_threshold=truth_sensitivity_indel_threshold)
+
+    sorted_tranche_data = hl.sorted(vd.tranche_data, key=lambda x: x.truth_sensitivity)
+    vd = vd.annotate_globals(snp_vqslod_threshold=
+                             sorted_tranche_data.filter(lambda x: (x.model == 'SNP') & (
+                                         x.truth_sensitivity >= truth_sensitivity_snp_threshold))
+                             .head()
+                             ,
+                             indel_vqslod_threshold=sorted_tranche_data.filter(lambda x: (x.model == 'INDEL') & (
+                                         x.truth_sensitivity >= truth_sensitivity_indel_threshold))
+                             .head()
+                             )
+
+    vd = vd.annotate_rows(
+        allele_NO=vd.alleles[1:].map(
+            lambda allele: hl.coalesce(vd.as_vqsr.get(allele).yng_status == 'N', False)),
+        allele_YES=vd.alleles[1:].map(
+            lambda allele: hl.coalesce(vd.as_vqsr.get(allele).yng_status == 'Y', True)),
+        allele_OK=vd.alleles[1:].map(
+            lambda alt: hl.coalesce(vd.as_vqsr.get(alt).vqslod > hl.if_else(hl.is_snp(vd.alleles[0], alt),
+                                                                            vd.snp_vqslod_threshold,
+                                                                            vd.indel_vqslod_threshold),
+                                    True)),
+    )
+
+    ft_criteria = (hl.range(vd.LGT.ploidy)
+                   .map(lambda idx: vd.LA[vd.LGT[idx]])
+                   .filter(lambda x: x != 0)
+                   .fold(lambda tuple, called_idx: hl.struct(
+        any_no=tuple[0] | vd.allele_NO[called_idx-1],
+        any_yes=tuple[1] | vd.allele_YES[called_idx-1],
+        all_ok_vqslod=tuple[2] & vd.allele_OK[called_idx-1])
+    , hl.struct(any_no=False, any_yes=False, all_ok_vqslod=True)))
```

Comment:
if 1 is a good allele and 2 is a bad allele, we want the following:

1/1  pass (true)
1/2 pass (true) <-- I dont think this is being handled correctly. We dont want to remove the one good allele, even if the other is bad
2/2 fail (false)
0/1 pass (true)
0/2 fail (false)

Reply:
I think we could take out "no" for now, since we dont have a use case for it

I think we would want this to be: all_no and any_yes and any_ok_vqslod
so the "|"s and "&"s would swap and the default values too